### PR TITLE
fix: [PL-40351]: Fixed the API Docs and Added custom setter

### PR DESCRIPTION
**Pull request by adiyaar24**
## Describe your changes
This PR Fixes the discrepancy in API Docs where the required fields were not marked as required , and also there were instance were we were using the builder to build it and those fields were not present , so added custom setter to it so that nothing breaks the flow 
Screenshot 2023-09-11 at 3 23 32 PM

The required fields were `size` , `K8sConfigDetails`, `delegateType`
## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52694)

### CODE REVIEW

**Comment by adiyaar24** [comment]
trigger messagecheck

**Comment by adiyaar24** [comment]
trigger uts

**Comment by adiyaar24** [comment]
trigger uts

**Comment by adiyaar24** [comment]
trigger uts

**Comment by adiyaar24** [comment]
trigger ut0

File: 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java [hunk 1]
```diff
@@ -2615,6 +2615,11 @@ public DelegateRegisterResponse register(final DelegateParams delegateParams, fi
             .projectIdentifier(projectIdentifier)
             .description(delegateParams.getDescription())
             .delegateType(delegateParams.getDelegateType())
+            .size(DelegateSize.SMALL)
```
Comment (AnupamIO) [inline 400-rest/src/main/java/software/wings/service/impl/DelegateServiceImpl.java:2618]: This is not correct, we will be override actual values with default values if we do this.
Example: If a delegate is installed with old flow using cluster_admin permission and xyz namespace. DelegateGroup collection will have below entry.
namespace: xyz
k8PermissionType : cluster_admin.

When delegate register will happen, actual value will get replaced by values you have hardcoded. i.e cluster_viewer and harness namespace.

**Pull request closed without merging**